test/testdata/infer/exception_loop.rb

```diff
@@ -11,17 +11,38 @@ def raises_b
   raise B.new
 end
 
-begin
-  raises_a
-rescue A => e
+def example1
+  begin
+    raises_a
+  rescue A => e
+  end
+
+  2.times do
+    begin
+      puts(e.class)
+      T.reveal_type(e) # error: `T.nilable(A)`
+      raises_b
+    rescue B => e
+      #         ^ error: Changing the type of a variable in a loop is not permitted
+    end
+  end
```

Comment:
Maybe make one or both of these a new CLI test so we have a check that the message gets triggered?  It's a helpful message and I'd hate to see it silently disappear because we changed locs for exception variables or something like that.

Reply:
oh I had meant to make a cli test, that’s for catching

Reply:
Oh I had meant to make a CLI test; thanks for catching